### ISSUE TITLE
Add `retry` option to `Deno.test`

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno#19882]
In the Fresh project we're running into more flaky tests on windows. Our tests are all based on `Deno.test` and it would be nice if we could retry failing tests automatically.

Something like:

```js
Deno.test({
  name: "foobar",
  retry: 3, // <-- Specifier number of times this test is allowed to fail.
  fn() {
     //...
  }
});
```

**Comment by israelidanny** [comment]
Wouldn't this kind of encourage writing of flaky tests? Feels like an anti-pattern of a feature to me, though I understand its' appeal when the test is flaky because of variables outside of your control.

**Comment by marvinhagemeister** [comment]
There is no obligation to set the `retry` option once implemented. Such an option is a must for end to end tests which are more flaky in nature due to the many variables outside of your control. We run into this exact problem daily over at the Fresh repository where we need to spin up a browser for tests.

**Comment by israelidanny** [comment]
Yes, you're absolutely right. For e2e / integration tests that include things like servers, dbs e.t.c this would be super helpful.

**Comment by lino-levan** [comment]
Would love this feature. Huge amount of support from me. I've been writing code with a lot of retrying to get around this recently.
